docs/service-contraints.md

```diff
@@ -0,0 +1,80 @@
+# Distbench node attributes and service constraints support
+
+Distbench node managers can have different attributes as part of their description, as
+well as a test can have service constraints referring to where the services can or cannot be placed.
+
+## Node manager attributes
+
+Before starting node_managers, we need to run the test_sequencer:
+```bash
+#(On 1 server node)
+bazel run :distbench -c opt -- test_sequencer --port=10000
+```
+
+To specify node attributes we need to run distbench on node_manager mode and add attributes through the command line with the format name=value. For instance, suppose we want to initialize a node with an attribute called rack with value 'A', we would run the following command:
+```bash
+#(On another server node)
+bazel run :distbench -c opt -- node_manager --test_sequencer=localhost:10000 --port=9999 rack=A
```

Comment:
This refers to localhost:10000 but says this is running on another machine. It's better to just leave this as a variable reference like --test_sequencer=$TEST_SEQUENCER_HOSTPORT